parsl/executors/ipp_controller.py

```diff
@@ -45,9 +46,17 @@ def __init__(self, public_ip=None, interfaces=None, port=None, port_range=None,
         self.interfaces = interfaces
         self.port = port
         self.port_range = port_range
+
         if port_range is not None:
             port_min, port_max = [int(x) for x in port_range.split(',')]
-            self.port = random.randint(port_min, port_max)
+            (
+              self.port,
+              self.hb_low, self.hb_high,
+              self.control_low, self.control_high,
+              self.mux_low, self.mux_high,
+              self.task_low, self.task_high
+            ) = random.sample(range(port_min, port_max), 9)
```

Comment:
We should do a sort on the random.sample to guarantee that the port_low < port_high for all pairs.

Reply:
That's a good point! I just realized low and high were bad terms to use though, these are just client/engine port  pairs, so the first doesn't need to be lower than the second in value.

Reply:
Okay, that makes sense :) Will merge once you have rename them.

Reply:
:+1: